docs/01-basic-usage.md

```diff
@@ -56,15 +58,19 @@ end
 
 The next arguments depend on the property we're getting. In this case, the third argument is the **origin**, which is a number from 0-1 that represents how far along the axis we should travel to find the point. So `get(name, 'x', 0)` would get the x position of the left edge of the box, `get(name, 'x', 0.5)` would get the x position of the horizontal center of the box, and `get(name, 'x', 1)` would get the x position of the right edge of the box. We can also use any origin in between.
 
-There's also some shortcut properties for getting x and y positions: `left`, `center`, and `right` return the left edge, horizontal center, and right edge of a box respectively, and `top`, `middle`, and `bottom` return the top edge, vertical center, and bottom edge of a box respectively. So we could write the above code like this:
+Note: Calling `layout:draw()` will clear the element tree of its state, and therefore attempting to call `get` with a named element will fail. In the above example, you can see we retrieve the `x` and `y` values _prior_ to calling `draw`.
```

Comment:
I'm unsure if you'd like this in here, or if you have something better instead? Without delving into the code, it was a little confusing as to why I couldn't use `get` _after_ I had drawn, so thought it useful to just highlight that here.

Reply:
I think that's a reasonable way to explain it - it correctly illuminates this confusing behavior that I should probably change....

Reply:
Wicked. Out of curiosity, why is it done this way? I mean, I don't know any better, so it seems reasonable to me haha.

Reply:
originally when the elements were only cleared upon the first `new` after a `draw`, if you didn't add any elements before the next `draw`, it would just re-draw what you had before. sometimes you legitimately don't want to draw anything, like if you're only drawing a shape conditionally, so it was confusing that it would keep drawing even after you stopped telling it to. i added a `begin` function to explicitly tell the layout to clear out the tree, but it's kind of a band-aid fix.